packages/cypress/src/add-commands/login.ts

```diff
@@ -0,0 +1,290 @@
+import { STORAGE_KEYS, OIDC_RESPONSE_TYPES } from '../constants';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+declare const Cypress: any;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+declare const cy: any;
+
+export type LoginCredentials = {
+  /**
+   * The user email.
+   * Defaults to `Cypress.env('LOGIN_EMAIL') ?? Cypress.env('LOGIN_USER')`.
+   */
+  email: string;
+  /**
+   * The user password.
+   * Defaults to `Cypress.env('LOGIN_PASSWORD')`.
+   */
+  password: string;
+};
+
+export type CommandLoginOptions = {
+  /**
+   * The application entry point URI path. This value is used to identify
+   * the correct application config.
+   */
+  entryPointUriPath: string;
+  /**
+   * Pass a list of dotfiles that must be loaded when the custom-application-config.json
+   * is loaded (in case you are using environment placeholder).
+   * By default the following dotfiles are loaded: `.env` and `.env.local`.
+   * You can also define the values using paths relative to the application folder.
+   */
+  dotfiles?: string[];
+  /**
+   * Called before your page has loaded all of its resources.
+   * Use this as a chance to interact for example with the browser storage.
+   */
+  onBeforeLoad?: (win: Window) => void;
+  /**
+   * If defined, visit this route after login.
+   */
+  initialRoute?: string;
+  /**
+   * The project key to access in the user session. The session token is valid for one project key at a time.
+   * Defaults to `Cypress.env('PROJECT_KEY')`.
+   */
+  projectKey?: string;
+  /**
+   * The user login credentials.
+   */
+  login?: LoginCredentials;
+};
+// Alias for backwards compatibility
+export type CommandLoginByOidcOptions = CommandLoginOptions;
+
+function loginByForm(commandOptions: CommandLoginOptions) {
+  if (isLocalhost()) {
+    throw new Error(
+      `At the moment, the "loginByForm" command only works when testing a Merchant Center production URL. Testing an application running on localhost is not supported due to issues with "cy.origin".`
+    );
+  }
+
+  const projectKey = commandOptions.projectKey ?? Cypress.env('PROJECT_KEY');
+
+  cy.task(
+    'customApplicationConfig',
+    {
+      entryPointUriPath: commandOptions.entryPointUriPath,
+      dotfiles: commandOptions.dotfiles,
+    },
+    // Do not show log, as it may contain sensible information.
+    { log: false }
+  ).then((appConfig: ApplicationRuntimeConfig['env']) => {
+    let url = `/${projectKey}/${commandOptions.entryPointUriPath}`;
+    if (commandOptions.entryPointUriPath === 'account') {
+      url = `/${commandOptions.entryPointUriPath}`;
+    }
+
+    // Log loaded application config for debugging purposes.
+    Cypress.log({
+      displayName: 'task',
+      name: 'customApplicationConfig',
+      message: appConfig,
+    });
+
+    const userCredentials = commandOptions.login ?? {
+      email: Cypress.env('LOGIN_EMAIL') ?? Cypress.env('LOGIN_USER'),
```

Comment:
**nit**: If user has something like this in his env file:
```
LOGIN_EMAIL=""
LOGIN_USER="sam@nowhere.com"
```
I would expect `LOGIN_USER` to be used but `LOGIN_EMAIL`.

This might happen in there are several `env` files (local, global, ...) and user is trying to override values.

Perhaps we can update this line to be like:
```suggestion
      email: Cypress.env('LOGIN_EMAIL') || Cypress.env('LOGIN_USER'),
```

Reply:
Good point, thanks for noticing!